qngng/qngng.py

```diff
@@ -244,21 +260,29 @@ class _Format(enum.Enum):
     CAP_CAMEL = 4
 
 
-def _format_name(fullname, fmt=_Format.DEFAULT):
+def _format_name(fullname, fmt=_Format.DEFAULT, middle_name_initial=None):
```

Comment:
Okay, about this: can you make another patch (before this one) which changes how this function works?

Instead of having multiple format strings for multiple formats, create a default full name and then pass it to `_normalize_name()` (if needed, depending on the format).

However, do not add the dot after the middle initial if the format is not `_Format.DEFAULT`.

So, for example, if the format is `_Format.KEBAB`:

1. Create this string:
       
       Guy A Lepage

2. Then pass it to `_normalize_name()` to get

       guy-a-lepage

What do you think?

Reply:
Of course the previous patch won't deal with a middle name/initial. This patch will add this.